tracer/src/Datadog.Trace/Debugger/Instrumentation/MethodDebuggerInvoker.cs

```diff
@@ -46,14 +47,25 @@ public static MethodDebuggerState BeginMethod_StartMarker<TTarget>(string probeI
                 return CreateInvalidatedDebuggerState();
             }
 
-            var state = new MethodDebuggerState(probeId, scope: default, DateTimeOffset.UtcNow, methodMetadataIndex, instance);
```

Comment:
Evaluating the condition should come before we make the rate-limiting decision, not after.